### PR TITLE
drivers: watchdog: add WDT shell commands

### PR DESCRIPTION
**Pull request by blemouzy**
Add shell commands to setup, disable, install timeout and feed a watchdog device.
These commands reflect watchdog API.

### CODE REVIEW

**Review by martinjaeger — DISMISSED** [review]

File: drivers/watchdog/wdt_shell.c [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+static int cmd_setup(const struct shell *sh, size_t argc, char *argv[])
+{
+	const struct device *dev;
+
+	dev = device_get_binding(argv[args_indx.device]);
+	if (!dev) {
+		shell_error(sh, "WDT device not found");
+		return -EINVAL;
```
Comment (kartben) [inline drivers/watchdog/wdt_shell.c:77]:
```suggestion
		return -ENODEV;
```

File: drivers/watchdog/wdt_shell.c [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+static int cmd_disable(const struct shell *sh, size_t argc, char *argv[])
+{
+	const struct device *dev;
+
+	dev = device_get_binding(argv[args_indx.device]);
+	if (!dev) {
+		shell_error(sh, "WDT device not found");
+		return -EINVAL;
```
Comment (kartben) [inline drivers/watchdog/wdt_shell.c:90]:
```suggestion
		return -ENODEV;
```

File: drivers/watchdog/wdt_shell.c [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	int flags;
+	int timeout_min;
+	int timeout_max;
+	struct wdt_timeout_cfg cfg;
+	int rc;
+
+	dev = device_get_binding(argv[args_indx.device]);
+	if (!dev) {
+		shell_error(sh, "WDT device not found");
+		return -EINVAL;
```
Comment (kartben) [inline drivers/watchdog/wdt_shell.c:108]:
```suggestion
		return -ENODEV;
```

File: drivers/watchdog/wdt_shell.c [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+static int cmd_feed(const struct shell *sh, size_t argc, char *argv[])
+{
+	const struct device *dev;
+	int channel_id;
+
+	dev = device_get_binding(argv[args_indx.device]);
+	if (!dev) {
+		shell_error(sh, "WDT device not found");
+		return -EINVAL;
```
Comment (kartben) [inline drivers/watchdog/wdt_shell.c:150]:
```suggestion
		return -ENODEV;
```

**Review by kartben — CHANGES_REQUESTED** [review]

**Review by aasinclair — COMMENTED** [review]
Thanks for this, it will be a great help when debugging watchdog issues.
Have you considered printing the **errno** when a function fails (see suggestion comment)?

File: drivers/watchdog/wdt_shell.c [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+	cfg.window.min = timeout_min;
+	cfg.window.max = timeout_max;
+	cfg.callback = NULL;
+	cfg.flags = flags;
+
+	rc = wdt_install_timeout(dev, &cfg);
+	if (rc >= 0) {
+		shell_print(sh, "Channel ID = %d", rc);
+	}
```
[the remark this answers is not in the document]
Comment (blemouzy) [reply drivers/watchdog/wdt_shell.c:137]: The `retval` command already exists to retrieve the last command return value.
So, it is really necessary?

**Review by aasinclair — APPROVED** [review]

**Review by martinjaeger — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**